thirdeye/thirdeye-frontend/app/pods/anomalies/controller.js

```diff
@@ -167,15 +173,13 @@ export default Controller.extend({
         // no filter applied, just return all
         return anomalyIds;
       }
-      let selectedAnomalies = [];
+      let selectedAnomalies = anomalyIds;
       filterMaps.forEach(map => {
-        if (anomalyFilters[map]) {
+        const selectedFilters = anomalyFilters[map];
+        // When a filter gets deleted, it leaves an empty array behind.  We need to treat null and empty array the same here
+        if (selectedFilters && Array.isArray(selectedFilters) && selectedFilters.length > 0) {
```

Comment:
You can use `isEmpty(selectedFilters)` here.

https://api.emberjs.com/ember/release/functions/@ember%2Futils/isEmpty

Reply:
Thank you -very useful!  I will push that change now.